pkg/adapters/grpc/faas_grpc.go

```diff
@@ -88,8 +88,10 @@ func (s *FaasServer) TriggerStream(stream pb.FaasService_TriggerStreamServer) er
 	log.Default().Println("FaaS stream closed, removing worker")
 
 	// Worker is done so we can remove it from the pool
-	s.pool.RemoveWorker(wrkr)
-
+	rwErr := s.pool.RemoveWorker(wrkr)
+	if err == nil {
+		err = rwErr
+	}
 	return err
```

Comment:
good that error handling is being added here, but the logic seems a little confusing, if we experience an error calling pool.RemoveWorker, but we had another error before, we're justing throwing away the new error?

Reply:
yeah, I wasn't sure what the priority was.. of if they were dependent. I could have an error list here..

Reply:
improvement comming